### ISSUE TITLE
Building with LWS_WITHOUT_TESTAPPS=ON should not generate SSH test keys

### ISSUE THREAD

**Issue by mulander** [warmcat/libwebsockets#1108]
I'm building with LWS_WITHOUT_TESTAPPS=ON and the configuration step generates the keys and the install step [unconditionally wants to install them](https://github.com/warmcat/libwebsockets/blob/v2.4-stable/CMakeLists.txt#L1862).

GENCERT is conditionally protected for LWS_WITHOUT_TEST_SERVER but not for LWS_WITHOUT_TESTAPPS (which when defined also disables the test server).

I believe extending this check:
```cmake
if (LWS_WITH_SSL AND OPENSSL_EXECUTABLE AND NOT LWS_WITHOUT_TEST_SERVER)
	set(GENCERTS 1)
```

with `AND NOT LWS_WITHOUT_TESTAPPS` will be enough to solve this issue and will proceed to test following up with a PR if I'm not wrong on the solution. I also don't see a reason to generate the keys when `LWS_WITHOUT_SERVER` doesn't make much sense as that stops the test server from building.

**Comment by mulander** [comment]
The keys are still unconditionally installed, unfortunately I think the condition needs to be repeated for the install command also.

**Comment by mulander** [comment]
https://github.com/warmcat/libwebsockets/pull/1109 updated with install conditional.

**Comment by lws-team** [comment]
Thanks it's pushed.  The logic got a bit mangled a while back for ESP32 case, it does not generate the test server but it needed to generate test certs for the user.  However I updated the ESP32 stuff recently to generate its own selfsigned cert at runtime if it sees it has no cert, so your changes are fine for that too.

**Comment by mulander** [comment]
Closing the issue as this was solved with 7a33b79ae4472b6f4631e428cdfaab2aeef8d632
